nir/src/main/scala/scala/scalanative/nir/serialization/BinaryDeserializer.scala

```diff
@@ -12,11 +14,7 @@ final class BinaryDeserializer(buffer: ByteBuffer) {
   private val header: Map[Global, Int] = {
     buffer.position(0)
 
-    val hasEntryPoint = getBool
-
-    val magic    = getInt
-    val compat   = getInt
-    val revision = getInt
+    val NirPrelude(magic, compat, revision, _) = NirPrelude.readFrom(buffer)
 
     assert(magic == Versions.magic, "Can't read non-NIR file.")
     assert(compat == Versions.compat && revision <= Versions.revision,
```

Comment:
These two assertions should be moved to `NirPrelude.readFrom`, *before* trying to read the byte for `existsEntryPoint`.

Reply:
The problem with this is that we don't have access to the `Versions` object in `util` and I am not very keen in making `util` depend on anything. I believe it should be dependency-free.

We could move `Versions` to `util` (as in `NirVersions`), but I don't really like that either. `Versions` should stay in `nir`, since they indicate NIR versions. On the other hand, since NIR is core to the project, there is no harm in its version being visible from the outside (for the modules that depend on `util`).

If I were to choose between the two above, I think that the second solution is preferable.

Reply:
Of course, we could provide the `Version` in the `NirPrelude.readFrom` call...

Reply:
What about moving `NirPrelude` to `nir/` instead? That seems like the obvious solution to me.

Reply:
True. I totally missed that `tools` depends on `nir`.